File: TTS (Text-To-Speech)/eleven-v3.py
import os
from elevenlabs.client import ElevenLabs
from dotenv import load_dotenv

load_dotenv(dotenv_path=".env.local")
client = ElevenLabs(api_key=os.getenv("ELEVENLABS_API_KEY"))
response = client.text_to_speech.convert(
  voice_id="21m00Tcm4TlvDq8ikWAM",
  model_id="eleven_v3",
  text="This is a test for the API of ElevenLabs."
)
with open("output.bin", "wb") as f:
  f.write(b"".join(response))
print("Saved output.bin")

File: TTS (Text-To-Speech)/eleven-v3.js
import fs from 'node:fs';
import process from "node:process";
import dotenv from "dotenv";
import { ElevenLabsClient } from "@elevenlabs/elevenlabs-js";

dotenv.config({ path: ".env.local" });

async function main() {
  const client = new ElevenLabsClient({
    apiKey: process.env.ELEVENLABS_API_KEY,
  });
  const response = await client.textToSpeech.convert(
    "21m00Tcm4TlvDq8ikWAM",
    {
      modelId: "eleven_v3",
      text: "Hey there, this is a test of Eleven v3 TTS model."
    }
  );
  fs.writeFileSync('output.bin', Buffer.from(response));
  console.log('Saved output.bin');
}
main();
